controllers/configurationpolicy_controller.go

```diff
@@ -860,29 +936,34 @@ func (r *ConfigurationPolicyReconciler) handleObjects(
 	objNames := []string{}
 	remediation := policy.Spec.RemediationAction
 
-	name, kind, metaNamespace := getDetails(unstruct)
-	if metaNamespace != "" {
-		namespace = metaNamespace
-		log = log.WithValues("namespace", metaNamespace)
+	// Since the object namespace is retrieved in the calling function, this condition should not be
+	// met and is an error
+	if objDetails.namespace != "" && objDetails.namespace != namespace {
```

Comment:
If this is true, then this is a programming error. Perhaps you could just `panic` in this case since it's impossible to reach unless we introduce a bug.